include/SFML/System/Err.hpp

```diff
@@ -29,16 +29,68 @@
 ////////////////////////////////////////////////////////////
 #include <SFML/System/Export.hpp>
 
-#include <iosfwd>
+#include <mutex>
+#include <ostream>
+#include <utility>
 
 
 namespace sf
 {
+////////////////////////////////////////////////////////////
+class SFML_SYSTEM_API ErrStream
+{
+private:
+    class Guard
+    {
+    public:
+        explicit Guard(std::ostream& stream, std::unique_lock<std::mutex>&& lockGuard);
+
+        Guard(const Guard&)            = delete;
+        Guard& operator=(const Guard&) = delete;
+
+        Guard(Guard&&)            = delete;
+        Guard& operator=(Guard&&) = delete;
+
+        Guard& operator<<(std::ostream& (*func)(std::ostream&));
```

Comment:
Why is it needed? Can't this be handled by the template operator<<?

Reply:
Nope. https://gcc.godbolt.org/z/zrsnoe7vf

Reply:
Oh, I think that is because `std::endl` is a template, ok.